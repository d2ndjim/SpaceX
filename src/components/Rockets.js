import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchRockets, updateReserved } from '../redux/rocketsSlice';
import classes from './Rockets.module.css';

const Rockets = () => {
  const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);

  const clickHandler = (id) => {
    dispatch(updateReserved(Number(id)));
  };

  return (
    <div>
      <div>
        {rockets.map((rocket) => (
          <div key={rocket.id} className={classes.container}>
            <img src={rocket.flickr_images} alt={rocket.rocket_name} className={classes.image} />
            <div className={classes.info}>
              <h2 className={classes.name}>{rocket.rocket_name}</h2>
              <p className={classes.desc}>
                {rocket.reserved && <span className={classes.reserved}>Reserved</span>}
                {rocket.description}
              </p>
              {rocket.reserved && (<button type="button" className={classes.btn} onClick={() => clickHandler(rocket.id)}>Cancel Reservation</button>)}
              {!rocket.reserved && (
              <button type="button" className={classes.btn} onClick={() => clickHandler(rocket.id)}>
                Reserve Rocket
              </button>
              )}
            </div>
          </div>
        ))}
      </div>
    </div>
  );
};

export default Rockets;
