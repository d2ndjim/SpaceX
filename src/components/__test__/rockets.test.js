import React from "react";
import renderer from 'react-test-renderer'
import { Provider } from "react-redux";
import store from '../../redux/configureStore'
import Rockets from '../Rocket/Rockets'

describe('Rockets test', () => {
  it ('renders rockets', () => {
    const app = renderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>
    ).toJSON();
    expect(app).toMatchSnapshot();
  })
})