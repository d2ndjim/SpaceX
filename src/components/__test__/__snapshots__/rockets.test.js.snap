// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets test renders rockets 1`] = `
<div>
  <div />
</div>
`;
